### DISCUSSION TITLE
Executing scripts in stimulus in html-button-response

### DISCUSSION THREAD

**Discussion by Yi-fengyu** [jspsych/jsPsych#3206] [Q&A] [answered]
hi,
I'm just getting started with jspsych, and wondering when I put a html as a stimulus into the html-button-response plugin, will the scripts on that page still be able to run? I wrote a page that switches to the specific image by clicking the corresponding button, but when I put it into the plugin, it doesn't respond after I click on it.
If not, can I only use the external-html plugin and bringing in the page as an external page. And then, is there any way to set a thing like trial_duration on that. 
thanks！！

**Comment by Shaobin-Jiang** [comment, marked as answer]
I should think not, as that is [specified by HTML](https://www.w3.org/TR/2008/WD-html5-20080610/dom.html#innerhtml0):

> script elements inserted using innerHTML do not execute when they are inserted

But you do not really need external-html. I mean, external html is generally for cases where seriously complex html is needed. But for a simple case like what you have described, you can simply set the basic html in `stimulus` and add whatever interactive function you might want in `on_load`.

Here's a simplified example:

```javascript
let trial = {
  type: jsPsychHtmlKeyboardResponse,
  stimulus:
    '<p id="target">1</p>' +
    '<input type="button" id="button-1" value="1">' +
    '<input type="button" id="button-2" value="2">' +
    '<input type="button" id="button-3" value="3">',
  on_load: function () {
    document.querySelector("#button-1").addEventListener("click", function () {
      document.querySelector("#target").innerHTML = 1;
    });

    document.querySelector("#button-2").addEventListener("click", function () {
      document.querySelector("#target").innerHTML = 2;
    });

    document.querySelector("#button-3").addEventListener("click", function () {
      document.querySelector("#target").innerHTML = 3;
    });
  },
};
```

**Reply by Yi-fengyu** [reply]
got it. thank you so much.
